File: src/main/java/org/moinex/repository/investment/InvestmentTargetRepository.java
/*
 * Filename: InvestmentTargetRepository.java
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 */

package org.moinex.repository.investment;

import java.util.List;
import java.util.Optional;
import org.moinex.model.investment.InvestmentTarget;
import org.moinex.util.enums.AssetType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface InvestmentTargetRepository extends JpaRepository<InvestmentTarget, Integer> {
    List<InvestmentTarget> findAllByIsActiveTrueOrderByAssetTypeAsc();

    Optional<InvestmentTarget> findByAssetTypeAndIsActiveTrue(AssetType assetType);

    boolean existsByAssetType(AssetType assetType);
}

File: src/main/kotlin/org/moinex/repository/investment/InvestmentTargetRepository.kt
/*
 * Filename: InvestmentTargetRepository.kt (original filename: InvestmentTargetRepository.java)
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 *
 * Migrate to Kotlin on 07/03/2026
 */

package org.moinex.repository.investment

import org.moinex.model.enums.AssetType
import org.moinex.model.investment.InvestmentTarget
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface InvestmentTargetRepository : JpaRepository<InvestmentTarget, Int> {
    fun findAllByIsActiveTrueOrderByAssetTypeAsc(): List<InvestmentTarget>

    fun findByAssetTypeAndIsActiveTrue(assetType: AssetType): InvestmentTarget?

    fun findByAssetType(assetType: AssetType): InvestmentTarget?
}
